Autopilot/AttitudeManager/SendInstructionsToSafety.hpp

```diff
@@ -31,9 +32,10 @@ void SendToSafety_Init(void);
 * Communicates the desired actuator commands encoded in the channelOut array over I2C to
 * the safety chip. Internally, this function performs a conversion from percentages to pwm values.
 * This function is non blocking and returns right away.
-* @param[in]		channelOut 		the array (size of 4 floats) of percentages each channel should be set to.
+* @param[in]		channel 		the output channel for PWM
+* @param[in]		percent 		the percentage between 0-100 that the PWM channel should be set to
 * @return							the error struct, containing all info about any errors that may have occured.
 */
-SendToSafety_error_t SendToSafety_Execute(float *channelOut);
+SendToSafety_error_t SendToSafety_Execute(int channel, int percent);
```

Comment:
Is the idea to initiate an I2C exchange every time any channel is updated ? Or are you thinking about storing the variables inside the module until all channels are ready then shipping them off all at once.

Reply:
So the way I understand interchip is that the I2C exchange is 'threaded', and the setPWM method just updates an array that stores the channel percentages. So safety and autopilot do their exchanges completely irrespective to when setPWM is called, which is smart cause then you don't have to wait for say grabber output when the PID output mode sends commands to the ailerons or something like that

Reply:
*Oh, the I2C exchange basically just sends this array full of channel percentages inside a struct lol

Reply:
Oh that makes a ton of sense and is super clean. Sweet!